src/services/expenses.js

```diff
@@ -0,0 +1,82 @@
+'use strict';
+
+let expenses = [];
+
+function clearAll() {
+  expenses = [];
+}
+
+function getAll(params = {}) {
+  let filteredExpenses = [...expenses];
+
+  if (params.userId) {
+    filteredExpenses = filteredExpenses.filter(expense => (
+      expense.userId === +params.userId
+    ));
+  }
+
+  if (params.from && params.to) {
+    filteredExpenses = filteredExpenses.filter(expense => {
+      const expenseDate = new Date(expense.spentAt).getTime();
+      const dateFrom = new Date(params.from).getTime();
+      const dateTo = new Date(params.to).getTime();
+
+      if (dateFrom <= expenseDate && dateTo >= expenseDate) {
+        return true;
+      }
+
+      return false;
+    });
+  }
+
+  if (params.categories) {
+    filteredExpenses = filteredExpenses.filter(expense => (
+      expense.category === params.categories
+    ));
+  }
+
+  return filteredExpenses;
+}
+
+function getById(expenseId) {
+  const foundExpenses = expenses.find(expense => (
+    expense.id === +expenseId
+  )) || null;
+
+  return foundExpenses;
+}
+
+function remove(expenseId) {
+  const filteredExpenses = expenses.filter(expense =>
+    (expense.id !== +expenseId));
+
+  return filteredExpenses;
+}
+
+function setFilteredExpenses(filteredExpenses) {
+  expenses = [...filteredExpenses];
+}
+
+function create({ userId, title, amount, category, note, spentAt }) {
+  const newId = expenses.length > 0
+    ? Math.max(...expenses.map(expense => expense.id)) + 1
+    : 0;
```

Comment:
Same for users, you can delete id=5, then create a new expense, and will have id=5 again, this is wrong, deleted ids should stay deleted.